geps/gep-1742.md

```diff
@@ -301,8 +301,87 @@ Could not find any HTTP specific timeouts. PRs welcomed. 😊
 
 ## API
 
-TBD.
+From the above analysis, it appears that most implementations are capable of supporting
+the configuration of simple downstream request timeouts on HTTPRoute rules. This is a
+relatively small addition (one new field) that would benefit many users.
+
+Because request timeouts are very much related to retry attempts, we also propose
+addressing issue: [#1731](https://github.com/kubernetes-sigs/gateway-api/issues/1731)
+by adding another field for retry conifguration in an HTTPRoute rule. This would also
+benefit many users and seems to be implementable by many (most?) Gateway implementations.
+
+### GO
+
+```go
+type HTTPRouteRule struct {
+	// Timeout for HTTP requests, disabled by default.
+	Timeout *time.Duration `json:"timeout,omitempty"`
+	// Retry policy for HTTP requests.
+	Retries *HTTPRetry `json:"retries,omitempty"`
+  // ...
+}
+
+type HTTPRetry struct {
+	// Number of retries to be allowed for a given request.
+	Attempts int32 `json:"attempts,omitempty"`
+	// Timeout per attempt for a given request, including the initial call and any retries.
+	// Default is the value of the HTTPRouteRule request Timeout.
+	PerTryTimeout *time.Duration `json:"per_try_timeout,omitempty"`
+	// Specifies the conditions under which retry takes place.
+	// One or more conditions can be specified using a ‘,’ delimited list.
+	RetryOn string `json:"retry_on,omitempty"`
+	// Flag to specify whether the retries should retry to other localities.
```

Comment:
This seems under-specified given there is no mention of "localities" anywhere else in the repo. Can we expand on this a bit.

Reply:
Added another section below.